### PR TITLE
fix: address subset of golangci-lint issues

### PR DESCRIPTION
**Pull request by sudo-suhas**
Address issues for the following linters:
- gofumpt
- gci
- errcheck
- gosec
- nakedret
- bodyclose
- errorlint
- govet
- noctx
- rowserrcheck
- sqlclosecheck
- gocritic: exitAfterDefer
- revive:
  - bare-return
  - deep-exit

### CODE REVIEW

File: internal/store/postgres/lineage_repository.go [hunk 1]
```diff
@@ -177,9 +179,9 @@ func (repo *LineageRepository) removeGraph(ctx context.Context, execer sqlx.Exec
 	return nil
 }
 
-func (repo *LineageRepository) compareGraph(current, new asset.LineageGraph) (toInserts, toRemoves asset.LineageGraph) {
+func (*LineageRepository) compareGraph(current, new asset.LineageGraph) (toInserts, toRemoves asset.LineageGraph) {
```
Comment (bsushmith) [inline internal/store/postgres/lineage_repository.go:182]: here - the named returns are still present. Expected ?

File: internal/store/postgres/lineage_repository.go [hunk 2]
```diff
@@ -283,18 +285,18 @@ func (repo *LineageRepository) buildDownstreamQuery(urn string, level int) (quer
 	return repo.buildRecursiveQuery(alias, nonRecursiveBuilder, recursiveBuilder)
 }
 
-func (repo *LineageRepository) buildRecursiveQuery(alias string, nonRecursiveBuilder, recursiveBuilder sq.SelectBuilder) (query string, args []interface{}, err error) {
-
+func (*LineageRepository) buildRecursiveQuery(alias string, nonRecursiveBuilder, recursiveBuilder sq.SelectBuilder) (
+	query string, args []interface{}, err error,
```
Comment (bsushmith) [inline internal/store/postgres/lineage_repository.go:289]: named returns still present

File: internal/store/postgres/lineage_repository.go (hunk 1 above)
Comment (sudo-suhas) [reply internal/store/postgres/lineage_repository.go:182]: Named returns are meant to be used as documentation when there are multiple values with identical types being returned. So this is intentional.

File: internal/store/postgres/lineage_repository.go (hunk 2 above)
Comment (sudo-suhas) [reply internal/store/postgres/lineage_repository.go:289]: Named returns are meant to be used as documentation when there are multiple values with types that don't immediately make their meaning clear. So this is intentional.

**Review by bsushmith — APPROVED** [review]

**Pull request merged**